crates/editor/src/editor.rs

```diff
@@ -4110,6 +4110,39 @@ impl Editor {
                 active_index: 0,
                 ranges: tabstops,
             });
+
+            // Gets whether the currently entered snipper ends with a surround
+            let snapshot = self.buffer.read(cx).snapshot(cx);
+            let selection = self.selections.first::<Point>(cx);
+            let start_column = snapshot.indent_size_for_line(selection.start.row).len;
```

Comment:
Since you're taking the start column of the line, this might break, in case the cursor is in column in which different auto-close brackets are available.

Consider inserting a snippet in a string, for example:

```rs
let foobar = r#'fn <foobar>(<arg1>, <arg2>) {
  println!("this is function <foobar>")
 }'#;
```

It's a made-up example, but it shows that inside the Rust raw-string-literal different brackets are available vs outside of it.